### ISSUE TITLE
[dev.icinga.com #11696] Allow definition of zone membership in endpoint object

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icinga2#4173]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/11696

**Created by twidhalm on 2016-04-28 09:14:19 +00:00**

Assignee: _(none)_
Status: _New_ 
Target Version: _(none)_
Last Update: _2016-11-18 13:59:38 +00:00 (in Redmine)_

    Backport?: Not yet backported
    Include in Changelog: 1

---

Currently the Icinga 2 config allows the definition of endpoints in zone objects but not the other way around.

Since Icinga Director uses Zones of type external\_object for the managment of Slaves this way of defining endpoint-zone relationships costs a lot of flexibility. Director knows of the zones but can not change them because they are external objects. If you want to add a Endpoint to the zone you'll have to add the endpoint to the Icinga 2 config files and import them as external\_objects to icinga director as well.

If the configuration allowed for defining zone membership on endpoints the endpoints could be defined and added to the correct zones in director.

In short: Please add an option to the endpoint object to set the zone this endpoint is in.

---

**Relations**:

* relates [#10943](https://dev.icinga.com/issues/10943)

**Comment by N-o-X** [comment]
Not something we will implement in the near feature, because the use cases are kinda limited. Most people that manage endpoints with the Director will also manage the zones there.
